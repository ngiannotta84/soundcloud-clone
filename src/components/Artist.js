import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "../styles/artist.css";
import { profile } from "../media/icons";

const Artist = ({ name, image }) => {
  return (
    <div className="artist">
      <Link to={`/profile/${name}`} className="artist__link">
        <img
          src={image || profile}
          alt={`album by ${name}`}
          className="artist__cover-art"
        />
        <h2 className="artist__name">{name}</h2>
      </Link>
    </div>
  );
};

Artist.defaultProps = {
  image: null,
};

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string,
};

export default Artist;
