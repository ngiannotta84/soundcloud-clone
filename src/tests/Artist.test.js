import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Artist from "../components/Artist";

describe("Artist", () => {
  const validProps = {
    name: "valid name",
    image: "valid url",
  };

  test("snapshot", () => {
    const { asFragment } = render(
      <Router>
        <Artist {...validProps} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe("tests", () => {
    beforeEach(() => {
      render(
        <Router>
          <Artist {...validProps} />
        </Router>
      );
    });

    test("renders correctly", () => {
      expect(screen.getByRole("img")).toHaveAttribute(
        "alt",
        `album by ${validProps.name}`
      );
      expect(screen.getByText(validProps.name)).toBeInstanceOf(
        HTMLHeadingElement
      );
      expect(screen.getByRole("link")).toHaveAttribute(
        "href",
        `/profile/${validProps.name}`
      );
    });
  });
});
