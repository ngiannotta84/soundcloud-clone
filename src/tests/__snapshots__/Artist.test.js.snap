// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist snapshot 1`] = `
<DocumentFragment>
  <div
    class="artist"
  >
    <a
      class="artist__link"
      href="/profile/valid name"
    >
      <img
        alt="album by valid name"
        class="artist__cover-art"
        src="valid url"
      />
      <h2
        class="artist__name"
      >
        valid name
      </h2>
    </a>
  </div>
</DocumentFragment>
`;
